imap_processing/mag/l0/mag_l0_data.py

```diff
@@ -90,19 +94,24 @@ class MagL0:
     PRI_FNTM: int
     SEC_COARSETM: int
     SEC_FNTM: int
-    VECTORS: bytearray
+    VECTORS: np.ndarray | str
 
     def __post_init__(self):
         """Convert Vectors attribute from string to bytearray if needed.
 
         Also convert encoded "VECSEC" (vectors per second) into proper vectors per
         second values
         """
-        if isinstance(self.VECTORS, str):
-            # Convert string output from space_packet_parser to bytearray
-            self.VECTORS = bytearray(
-                int(self.VECTORS, 2).to_bytes(len(self.VECTORS) // 8, "big")
-            )
+        # Convert string output from space_packet_parser to numpy array of
+        # big-endian bytes
+        self.VECTORS = np.frombuffer(
+            int(self.VECTORS, 2).to_bytes(len(self.VECTORS) // 8, "big"),
+            dtype=np.dtype(">b"),
+        )
+
+        # Remove buffer from end of vectors
+        if len(self.VECTORS) % 2:
```

Comment:
What does this `% 2` indicate?

Reply:
The length of the vectors has to be an even number. If theres any number left over from %2, meaning self.Vectors is not an even number, then there is a buffer at the end that needs to be removed. I'm not totally sure _why_ there is a buffer, but in some test data the vector length would be 201 with a zero padding at the end. I think it has to do with CCSDS weirdness, it's present in their packet spreadsheet.